### PR TITLE
Post SparkOperationEvent for all SparkOperations and show sessionId for statements

### LINKED ISSUE #2886: [Bug]  kyuubi UI cann't show spark-scala-core statement running state
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-kyuubi/issues?q=is%3Aissue) and found no similar issues.

### Describe the bug

**Thies is my test code:**
     ` execute(conn, st, "create or replace temporary view orders_view as select * from ods_pupu_main.orders where status <> 10;"); `
       ` execute(conn, st, "set kyuubi.operation.language=scala;");`
        `String core = "val df = spark.sql(\"select id, pay_type from orders_view\")  \n" +
                "df.filter(\"pay_type <> -1\").createOrReplaceTempView(\"res\") \n" +
                "df.count()  \n";
        execute(conn, st, core);`
        `execute(conn, st, "spark.conf.set(\"kyuubi.operation.language\", \"sql\")");`
        `execute(conn, st, "select * from res limit 3");`

**The kyuubi UI only show  sql-running-state no core-running-state.**

### Affects Version(s)

1.5.1

### Kyuubi Server Log Output

_No response_

### Kyuubi Engine Log Output

_No response_

### Kyuubi Server Configurations

_No response_

### Kyuubi Engine Configurations

_No response_

### Additional context

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by turboFei**
### _Why are the changes needed?_

Before, we can only see sql query on the kyuubi query engine tab.
image

After this pr, we can see both scala & python code in the kyuubi query engine tab.

image

Also support to show sessionId.
image

It also close #2886
### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [x] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/ExecutePython.scala [hunk 1]
```diff
@@ -167,6 +172,11 @@ class ExecutePython(
       }
     }
   }
+
+  override protected def setState(newState: OperationState): Unit = {
```
Comment (cfmcgrady) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/ExecutePython.scala:176]: Why not override method `setState` on the superclass `SparkOperation`, then all the operations can be seen in the tab. cc @cxzl25

Comment (cxzl25) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/ExecutePython.scala:176]: It sounds pretty good.

This PR should also fix https://github.com/apache/kyuubi/issues/2886

**Comment by turboFei** [comment]
cc @cfmcgrady @pan3793 @cxzl25 @wForget

**Review by cxzl25 — APPROVED** [review]
LGTM

**Review by ulysses-you — APPROVED** [review]

**Comment by turboFei** [comment]
thanks all for the review, merging to master

**Comment by cfmcgrady** [comment]
late lgtm

**Pull request closed without merging**